### PR TITLE
chore(dev): Install dd-rust-license-tool from crates.io

### PR DESCRIPTION
**Pull request by jszwedko**
And update vdev as well as the docs to use the new binary name.

Signed-off-by: Jesse Szwedko <jesse.szwedko@datadoghq.com>

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]
You updated the description just before I could mention it 🙂

File: scripts/environment/prepare.sh [hunk 1]
```diff
@@ -17,8 +17,8 @@ fi
 if ! cargo deny --version >& /dev/null ; then
   rustup run stable cargo install cargo-deny --force --locked
 fi
-if ! rust-license-tool --help >& /dev/null ; then
-  cargo install --git https://github.com/DataDog/rust-license-tool
+if ! dd-rust-license-tool --help >& /dev/null ; then
+  rustup run stable cargo install dd-rust-license-tool --version 1.0.1 --force --locked
```
Comment (bruceg) [inline scripts/environment/prepare.sh:21]: Do we need to lock down the version? We don't on other tools.

Comment (jszwedko) [reply scripts/environment/prepare.sh:21]: We do actually lock down all of the others (see the lines above this one) except `cargo-deny`, which I'd argue we should be locking too.

Comment (bruceg) [reply scripts/environment/prepare.sh:21]: LOL took the line above to be exemplary. Fair enough.

**Review by bruceg — APPROVED** [review]

**Pull request merged**